test/integration/istio_http_integration_test.cc

```diff
@@ -0,0 +1,328 @@
+      report_cluster: %s
+      check_cluster: %s
+  )";
+  return fmt::sprintf(kMixerFilterTemplate, kDestinationUID, kSourceUID, kTelemetryBackend, kPolicyBackend);
+}
+
+class IstioHttpIntegrationTest : public HttpProtocolIntegrationTest {
+ public:
+  void createUpstreams() override {
+    HttpProtocolIntegrationTest::createUpstreams();
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
+    telemetry_upstream_ = fake_upstreams_.back().get();
+
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
+    policy_upstream_ = fake_upstreams_.back().get();
+  }
+
+  void SetUp() override {
+    config_helper_.addFilter(MakeMixerFilterConfig());
+    config_helper_.addFilter(MakeAuthFilterConfig());
+    config_helper_.addFilter(MakeJwtFilterConfig());
+
+    config_helper_.addConfigModifier(addCluster(kTelemetryBackend));
+    config_helper_.addConfigModifier(addCluster(kPolicyBackend));
+
+    HttpProtocolIntegrationTest::initialize();
+  }
+
+  void TearDown() override {
+    cleanupConnection(fake_upstream_connection_);
+    cleanupConnection(telemetry_connection_);
+    cleanupConnection(policy_connection_);
+  }
+
+  ConfigHelper::ConfigModifierFunction addCluster(const std::string& name) {
+    return [name](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+      auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+      cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      cluster->mutable_http2_protocol_options();
+      cluster->set_name(name);
+    };
+  }
+
+
+  void waitForTelemetryRequest(::istio::mixer::v1::ReportRequest* request) {
+    AssertionResult result =
+        telemetry_upstream_->waitForHttpConnection(*dispatcher_, telemetry_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = telemetry_connection_->waitForNewStream(*dispatcher_, telemetry_request_);
+    RELEASE_ASSERT(result, result.message());
+
+    result = telemetry_request_->waitForGrpcMessage(*dispatcher_, *request);
+    RELEASE_ASSERT(result, result.message());
+  }
+
+  // Must be called after waitForTelemetryRequest
+  void waitForTelemetryResponse() {
+    telemetry_request_->startGrpcStream();
+    telemetry_request_->sendGrpcMessage(::istio::mixer::v1::ReportResponse{});
+    telemetry_request_->finishGrpcStream(Grpc::Status::Ok);
+  }
+
+  void waitForPolicyRequest(::istio::mixer::v1::CheckRequest* request) {
+    AssertionResult result =
+        policy_upstream_->waitForHttpConnection(*dispatcher_, policy_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = policy_connection_->waitForNewStream(*dispatcher_, policy_request_);
+    RELEASE_ASSERT(result, result.message());
+
+    result = policy_request_->waitForGrpcMessage(*dispatcher_, *request);
+    RELEASE_ASSERT(result, result.message());
+  }
+
+  // Must be called after waitForPolicyRequest
+  void waitForPolicyResponse() {
```

Comment:
call it  sendPolicyResponse()?  not waitFor.
same for sendTelemetryResponse()

Reply:
Done